applications/tari_console_wallet/src/ui/state/wallet_event_monitor.rs

```diff
@@ -160,6 +160,7 @@ impl WalletEventMonitor {
                                 match (*msg).clone() {
                                     BaseNodeEvent::BaseNodeStateChanged(state) => {
                                         self.trigger_base_node_state_refresh(state).await;
+                                        self.trigger_balance_refresh_check().await;
```

Comment:
I don't understand what this is for. This is not an event that would result in the balance changing?

Reply:
If that is then this should rather be in the on_tick event

Reply:
Please see above

Reply:
I still think this should be in the on_tick event of the TUI. Here you are linking this to Base Node Event which has nothing to do with balance.

Reply:
Agreed, done.